### PR TITLE
feat(integration): add a global handler for Literal AI

### PR DESCRIPTION
**Pull request by Dam-Buty**
# Description

This PR introduces a new `global_handler` integration with Literal AI. The code here, as well as the corresponding code in the Literal AI Python SDK, are based on the new `instrumentation` module.

Also included is a small update to the relevant bit in the Observability section of the documentation.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

I added an example script and hacked at it until it worked :thinking: 

- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/callbacks/llama-index-callbacks-literalai/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# LlamaIndex Callbacks Integration: Literalai
```
Comment (desaxce) [inline llama-index-integrations/callbacks/llama-index-callbacks-literalai/README.md:1]: Maybe put Literal AI

**Review by masci — CHANGES_REQUESTED** [review]
Thanks for the contribution! Do you mind filling the PR description explaining what the change is to help me review this? TIA!

File: docs/docs/module_guides/observability/index.md [hunk 2]
```diff
@@ -119,7 +119,10 @@ llama_index.core.set_global_handler("arize_phoenix")
 
 ### Literal AI
 
-[Literal AI](https://literalai.com/) is the go-to LLM evaluation and observability solution, enabling engineering and product teams to ship LLM applications reliably, faster and at scale. This is possible through a collaborative development cycle involving prompt engineering, LLM observability, LLM evaluation and LLM monitoring. Conversation Threads and Engine Runs can be automatically logged on Literal AI.
+[Literal AI](https://literalai.com/) is the go-to LLM evaluation and observability solution, enabling engineering and product teams to ship LLM applications reliably, faster and at scale. This is possible through a collaborative development cycle involving prompt engineering, LLM observability, LLM evaluation and LLM monitoring. Conversation Threads and Agent Runs can be automatically logged on Literal AI.
+
+The simplest way to get started and try out Literal AI is to signup on our [cloud instance](https://cloud.getliteral.ai/).
+You can then navigate to **Settings**, grab your API key, and start logging !
```
Comment (masci) [inline docs/docs/module_guides/observability/index.md:125]:
```suggestion
You can then navigate to **Settings**, grab your API key, and start logging!
```

File: docs/docs/module_guides/observability/index.md [hunk 3]
```diff
@@ -130,18 +133,11 @@ llama_index.core.set_global_handler("arize_phoenix")
...
-# LITERAL_API_KEY
-# LITERAL_API_URL
-literalai_client = LiteralClient()
-
-# You can also explicitly provide the configuration keys like so :
-# literalai_client = LiteralClient(api_key="lsk_xxxxx", url="http://localhost:3000")
+from llama_index.core import set_global_handler
 
-literalai_client.instrument_llamaindex()
+# You should provide your Literal AI API key and base url using the following environment variables :
```
Comment (masci) [inline docs/docs/module_guides/observability/index.md:138]:
```suggestion
# You should provide your Literal AI API key and base url using the following environment variables:
```

**Review by masci — CHANGES_REQUESTED** [review]
PR looks great, I've only found two almost invisible typos in the docs 😬

**Comment by Dam-Buty** [comment]
Thanks for the keen eye @masci ! Both suggestions applied

**Pull request merged**